packages/data-models/flowTypes.ts

```diff
@@ -2,7 +2,7 @@
 
 import type { IDataPipeOperation } from "shared";
 import type { IAppConfig } from "./appConfig";
-import { IAssetEntry } from "./deployment.model";
+import type { IAssetEntry } from "./deployment.model";
```

Comment:
Yeah, I've worked on codebases before where they enforce lint rules to ensure all types are imported in this way. The main impact depends on what else is in the file you are importing. The reason I needed type imports here was that during some test development the `flowTypes.ts` file also had reference to code that isn't accessible to browser frontend app (either some code that only works in the node environment, e.g. `fs` references, or as part of a workspace that isn't imported, e.g. `scripts`)

By importing a type only then you don't have to worry about the chained imports that could accidentally be pulled in. In an ideal world there'd be better separation between type definitions and functions so that you just import files from types that only reference types, however I'm often lazy an include `index.ts` files that bulk export from multiple files, and inline the odd type definition alongside classes when first coding (forgetting that it might be needed by other code in the future).